runtime/buffer.cc

```diff
@@ -272,7 +272,7 @@ void copy_impl(raw_buffer& src, raw_buffer& dst, const void* padding) {
     constant_buffer buffer;
     if (padding) {
       optimize_fill_value(padding, elem_size, buffer);
-    } else if (src.dim(0).empty()) {
+    } else if (src.rank > 0 && src.dim(0).empty()) {
```

Comment:
I'm not sure about this. Up above, we have `assert(src.rank == dst.rank)`, and then we have `if (rank == 0) { ... }` and handle that case. So I think by getting here, we've skipped that assert? Or is something else going on here? Maybe a bug in `optimize_dims` that is fusing a different number of dimensions in `src` and `dst`?

If the `assert(src.rank == dst.rank)` isn't failing just due to running without asserts, then I'd add another `assert(src.rank == dst.rank)` after the `optimize_dims` call.

Reply:
> We should also test a rank 2 src with a non-empty dim 0 and empty dim 1, but I'm pretty sure that should already be working.

With the proposed fix in place, empty-dim-0 works but empty-dim-1 segfaults inside fill(), debugging now